### PR TITLE
#768 : [1.1-develop] Introduce MediaGalleryUi module

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR will introduce MediaGalleryUi module which will be responsible for Media gallery user interface
### Fixed Issues (if relevant)

1. https://github.com/magento/adobe-stock-integration/issues/768: [1.1-develop] Introduce MediaGalleryUi module

### Manual testing scenarios (*)

1. Login into Magento Backend
2. Check the configuration in in Stores -> Configuration -> Advanced -> System -> Media Gallery fieldset

### Screenshot

image

### CODE REVIEW

**Comment by konarshankar07** [comment]
Hello @sivaschenko ,
Do we also need to implement the data service contract for `config.php` in `MediaGalleryUiApi` module?

File: MediaGalleryUi/Test/Unit/Model/ConfigTest.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Get target environment test.
+     */
+    public function testIsEnabled(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('isSetFlag')
+            ->with(self::XML_PATH_ENABLED)
+            ->willReturn(true);
+        $this->config->isEnabled();
```
Comment (sivaschenko) [inline MediaGalleryUi/Test/Unit/Model/ConfigTest.php]: This test does not actually test the result of `isEnabled` method call. I'd use a data provider to check both `true` and `false` values and add an assertion:
```suggestion
        $this->assertEquals(true, $this->config->isEnabled());
```

File: MediaGalleryUi/Model/Config.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Class Config
+ */
+class Config
+{
+    /**
+     * Path to enable/disable media gallery in the system settings.
+     */
+    private const XML_PATH_ENABLED = 'media/gallery/enabled';
```
Comment (sivaschenko) [inline MediaGalleryUi/Model/Config.php]: Please update the path to (also affects the unit test, config.xml and system.xml):
```suggestion
    private const XML_PATH_ENABLED = 'system/media_gallery/enabled';
```

File: MediaGalleryUi/Model/Config.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+     *
+     * @param ScopeConfigInterface $scopeConfig
+     */
+    public function __construct(ScopeConfigInterface $scopeConfig)
+    {
+        $this->scopeConfig = $scopeConfig;
+    }
+
+    /**
+     * Is media gallery enabled
```
Comment (sivaschenko) [inline MediaGalleryUi/Model/Config.php]:
```suggestion
     * Check if masonry grid UI is enabled for Magento media gallery
```

File: MediaGalleryUi/Model/Config.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model;
+
+use Magento\Framework\App\Config\ScopeConfigInterface;
+
+/**
+ * Class Config
```
Comment (sivaschenko) [inline MediaGalleryUi/Model/Config.php]:
```suggestion
 * Class responsible to provide access to system configuration related to the Media Gallery
```

File: MediaGalleryUi/composer.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "name": "magento/module-media-gallery-ui",
+    "description": "Magento module responsible for the media gallery UI implementation",
+    "require": {
+        "php": "~7.1.3||~7.2.0||~7.3.0",
+        "magento/framework": "*",
+        "magento/module-backend": "*",
+        "magento/module-ui": "*"
+    },
+    "suggest": {
+        "magento/module-cms": "*"
```
Comment (sivaschenko) [inline MediaGalleryUi/composer.json]: I think we can suggest installing "magento/module-media-gallery", but not "magento/module-cms".

File: MediaGalleryUi/composer.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+    "name": "magento/module-media-gallery-ui",
+    "description": "Magento module responsible for the media gallery UI implementation",
+    "require": {
+        "php": "~7.1.3||~7.2.0||~7.3.0",
+        "magento/framework": "*",
+        "magento/module-backend": "*",
+        "magento/module-ui": "*"
```
Comment (sivaschenko) [inline MediaGalleryUi/composer.json]: There is no dependency on backend and ui modules currently

File: MediaGalleryUi/modman [hunk 7]
```diff
@@ -0,0 +1 @@
+./ app/code/Magento/MediaGalleryUi
```
Comment (sivaschenko) [inline MediaGalleryUi/modman:1]: Global modman file should be updated as well

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @konarshankar07 thanks for the pull request! Congratulations on creating a first PR to 1.1 version!

Please take a look at my review comments.

As for `ConfigInterface` in `MediaGalleryUiApi`- I moved that work to a separate task https://github.com/magento/adobe-stock-integration/issues/774

File: MediaGalleryUi/etc/adminhtml/system.xml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
+    <system>
+        <section id="system">
+            <group id="media_gallery" translate="label" type="text" sortOrder="1000" showInDefault="1" showInWebsite="0" showInStore="0">
+                <label>Media Gallery</label>
+                <field id="enabled" translate="label" type="select" sortOrder="10" showInDefault="1" showInWebsite="0" showInStore="0">
+                    <label>Enabled Media Gallery</label>
```
Comment (sivaschenko) [inline MediaGalleryUi/etc/adminhtml/system.xml]: No need to specify "Media Gallery" both in fieldset and field names
```suggestion
                    <label>Enabled</label>
```

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**